### PR TITLE
Added threshold check for each rule from config.yml file

### PR DESCRIPTION
**Pull request by makaroni4**

### CODE REVIEW

**Comment by makaroni4** [comment]
@mgrecar @dkarter could you please review this PR? This is the implementation of thresholds for each rule via `config.yml` or CLI `-t 90,90,90,90`

**Comment by makaroni4** [comment]
#40

File: spec/test_helper.rb [hunk 1]
```diff
@@ -6,12 +6,14 @@
 RSpec.configure do |config|
   original_stderr = $stderr
   original_stdout = $stdout
-  config.before(:all) do
+
+  config.before(:all, silent_cli: true) do
```
Comment (dkarter) [inline spec/test_helper.rb]: So if you are adding the silent_cli attribute do you still want to keep `:all`?

Comment (makaroni4) [inline spec/test_helper.rb]: Very good point, fixed :+1:

Comment (dkarter) [inline spec/test_helper.rb]: I'm not sure if what I said was right. you may still need `:all`. Did it work ok (output from cli silenced) when you removed `:all`?

Comment (makaroni4) [inline spec/test_helper.rb]: Sure, no param means `:each` option with `:silent_cli` tag. Specs are OK, Travis will build it soon.

**Comment by dkarter** [comment]
:+1:

**Pull request merged**